### ISSUE TITLE
pyright ignores pythonPath from folder settings

### ISSUE THREAD

**Issue by ironfoot** [microsoft/pyright#166]
**Describe the bug**
VSCode allows having per folder settings that can override workspace setting. When pyright runs inside VSCode, it uses the python interpreter from the workspace settings and ignores the one set in folders.

**To Reproduce**
1. Prepare two python interpreter using conda. (or any other virtualenv like tool)
    I prepared two conda env, one with pytest installed and other without.
1. Create a workspace in VSCode.
1. Select interpreter without pytest. It gets added to User settings by default. It doesn't really matter for this repro. (the result is same if I move `python.pythonPath` is user settings to workspace settings.)
1. Add an empty folder to the workspace.
1. Select interpreter with pytest. It gets added to folder settings. My folder setting looks like this at this point.
    ```json
    {
        "python.pythonPath": "/Users/ironfoot/miniconda3/envs/repro/bin/python"
    }
    ```
1. Create "test.py"
    ```python
    import pytest
    # pyright detects "Import 'pytest' could not be resolved"
    ```
1. Set interpreter with pytest at workspace level. My workspace setting looks like this at this point.
    ```json
    {
        "folders": [
            {
                "path": "/Users/ironfoot/workspace/repro"
            }
        ],
        "settings": {
            "python.pythonPath": "/Users/ironfoot/miniconda3/envs/repro/bin/python"
        }
    }
    ```
1. Now pyright correctly detects pytest in "test.py"

The real world application is that I have a multi-root workspace. Each folder has it's own python requirements that differ from each other. I can't get pyright to work in both folders.
- If I don't set interpreter at workspace level, pyright doesn't work in any folder.
- If I set choose one interpreter and put it at workspace level, pyright works on that folder works but not others.

**Expected behavior**
pyright should respect `python.pythonPath` in folder settings.
When running pyright inside files in a folder, pyright should use interpreter set in that folder's settings.

**VS Code extension or command-line**
VSCode extension. 1.0.44

**Comment by erictraut** [comment]
I think this is a duplicate of #125.

**Comment by ironfoot** [comment]
I'm guessing folder settings is a part of multi-root workspace? So, by adding a single folder to the workspace through "Add folder to workspace" menu, I've essentially created a multi-root workspace where N=1?

Just tried "open" a folder from an empty workspace instead of "add" one folder to an empty workspace and no folder settings was created and interpreter selection went straight into workspace settings. And that means pyright picks the correct interpreter.

So... yes. It seems like a duplicate to #125

**Comment by erictraut** [comment]
I've added basic multi-root workspace in pyright. It now supports different pyrightconfig.json files within each root. Settings specified via VS Code (including python.pythonPath) are handled globally, not per-folder.

I'll leave this feature request open for now and further investigate adding support for per-folder VS Code settings.

**Comment by erictraut** [comment]
I've added support for per-folder settings in multi-root workspaces. This will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is fixed in 1.0.47, which I just published.
